File: PS/algo/src/BOJ/Silver/BOJ14916.java
package BOJ.Silver;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class BOJ14916 {
    public static void main(String[] args) throws IOException {
        BufferedReader br =new BufferedReader(new InputStreamReader(System.in));
        int n =Integer.parseInt(br.readLine());
        int count =0;
        while(n>0){
            if(n%5!=0){
                n-=2;
                count++;
            }else{
                n-=5;
                count++;
            }
        }
        System.out.println(n<0?-1:count);
    }
}

File: src/main/kotlin/BOJ/silver/BOJ14916.kt
package BOJ.silver

import java.io.BufferedReader

fun main() = with(BufferedReader(System.`in`.bufferedReader())) {
    val n  = readLine().toInt()
    val answer = check(n)
    println(answer)
}

tailrec fun check(n:Int ,count:Int=0): Int {
    return if(n==0){
        count
    } else if(n<0){
        -1
    } else {
        if(n%5==0) check(n-5,count+1)
        else check(n-2,count+1)
    }
}